include/hipCPU/hip/detail/queue.hpp

```diff
@@ -170,9 +179,11 @@ void async_queue::work()
     _is_idle = false;
     operation();
 
-    if(_enqueued_operations.empty())
-      _is_idle = true;
-
+    _is_idle = [this] {
+      std::lock_guard<std::mutex> lock(_mutex);
+      return _enqueued_operations.empty();
+    }();
```

Comment:
I wonder if it might be clearer to just have 
```c++
{
  std::local_guard<std::mutex> lock(_mutex);
  _is_idle = _enqueued_operations.empty();
}
```
?

Reply:
Sure, you're right, I was wondering how to sync the access to `_enqueued_operations` only since `_is_idle` is atomic anyway and overlook the overly complicated solution I used :)